main.go

```diff
@@ -244,6 +249,39 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "Mirroring")
 		os.Exit(1)
 	}
+
+	// shared event channel and sync map between the scraper and the reconciler
+	eventCh := make(chan event.GenericEvent)
+	syncMap := &sync.Map{}
+
+	if err = mgr.Add(
+		manager.RunnableFunc(func(ctx context.Context) error {
+			// run a go routine to scrape the metrics from prometheus periodically
+			go (&storageautoscaler.StorageAutoscalerScraper{
```

Comment:
Define the var on its own line, then on the next line call `go scraper.ScrapteMetricsPeriodically(ctx)`. I know it's possible this way, but for such a complex struct it will be more readable that way.